### PR TITLE
DBZ-1643 Manage sql server timestamp timezone

### PR DESCRIPTION
**Pull request by jpechane**
Added test

### CODE REVIEW

**Comment by gunnarmorling** [comment]
@jvenant, @jpechane, I've pushed one more commit to this branch and I believe it's working correctly finally.

The case where the incoming value already is converted to UTC by means of `AT TIME ZONE` (as possible on SQL Server 2016 and newer) is covered by the test that Jiri added.

The other case is hard to test in an automated way, as we don't have a container image for 2014. So I tried to emulate that behavior by changing the statement in `getLsnToTimestamp()` so it contains `AT TIME ZONE 'US Eastern Standard Time'`. Then when I set `database.server.timezone` to the corresponding value `US/Eastern`, the final timestamp value has the correct UTC representation and thus millis since epoch. The test passes in this case, too.

File: debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java [hunk 1]
```diff
@@ -379,12 +394,22 @@ public String getRealDatabaseName() {
     private ZoneId retrieveTransactionTimezone(boolean supportsAtTimeZone) {
         final String serverTimezoneConfig = config().getString(SERVER_TIMEZONE_PROP_NAME);
 
-        if (!supportsAtTimeZone && serverTimezoneConfig == null) {
-            LOGGER.warn("The ' " + SERVER_TIMEZONE_PROP_NAME
-                    + "' option should be specified to avoid incorrect timestamp values in case of different timezones between the database server and this connector's JVM.");
+        if (supportsAtTimeZone) {
+            if (serverTimezoneConfig != null) {
+                LOGGER.warn("The '{}' option should not be specified with SQL Server 2016 and newer");
```
Comment (jpechane) [inline debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java]: Missing option name in the log

File: debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java [hunk 2]
```diff
@@ -379,12 +394,22 @@ public String getRealDatabaseName() {
...
+            }
+        }
+        else {
+            if (serverTimezoneConfig == null) {
+                LOGGER.warn("The ' " + SERVER_TIMEZONE_PROP_NAME
```
Comment (jpechane) [inline debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java]: You should make log messages consistent, either both using slf4j placeholders or bot using string concats but not both

**Comment by jpechane** [comment]
@gunnarmorling LGTM, just 2 nitpicks regardng logging added

File: debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java (hunk 2 above)
Comment (gunnarmorling) [reply debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java]: Good catch on both. I'll adjust and merge. Thanks for reviewing, @jpechane!

**Comment by gunnarmorling** [comment]
Applied. Thanks, all. This will be in tomorrow's release 1.0 Final release, @jvenant.

**Comment by jvenant** [comment]
\o/
Thank you very much guys !

**Comment by maurolscla** [comment]
It seems that this has been taken care of regarding the transaction timestamp (`ts_ms`). Is this the case for any other field/column in database tables as well? 
The behaviour I'm seeing on my side is that `datetime` fields are being converted to UTC timestamps regardless of both SQLServer's or kafka-connect worker's timezone.
For instance:
My database is set to PDT/PST (UTC-7) and `2020-09-17 11:22:51` (my time) is being converted to `1600341771897` which is `2020-09-17 11:22:51` at UTC.
Is this expected behaviour?

**Pull request closed without merging**